src/hotspot/share/gc/shenandoah/shenandoahLock.cpp

```diff
@@ -28,9 +28,47 @@
 
 #include "gc/shenandoah/shenandoahLock.hpp"
 #include "runtime/atomic.hpp"
+#include "runtime/interfaceSupport.inline.hpp"
 #include "runtime/os.inline.hpp"
 #include "runtime/thread.hpp"
 
+// These are inline variants of Thread::SpinAcquire with optional blocking in VM.
+
+class ShenandoahNoBlockOp : public StackObj {
+public:
+  ShenandoahNoBlockOp(JavaThread* java_thread) {
+    assert(java_thread == NULL, "Should not pass anything");
+  }
+};
+
+void ShenandoahLock::contended_lock(bool allow_block_for_safepoint) {
+  Thread* thread = Thread::current();
+  if (allow_block_for_safepoint && thread->is_Java_thread()) {
+    contended_lock_internal<ThreadBlockInVM>(static_cast<JavaThread*>(thread));
```

Comment:
The preferred style in older JDKs is `thread->as_Java_thread()`.

Reply:
Thanks, I have updated code to use thread->as_Java_thread() instead.